activerecord/lib/active_record/migration.rb

```diff
@@ -574,15 +574,15 @@ def nearest_delegate # :nodoc:
       end
 
       # Raises <tt>ActiveRecord::PendingMigrationError</tt> error if any migrations are pending.
-      def check_pending!(connection = Base.connection)
+      def check_pending!(connection = self.class.connection)
```

Comment:
This method itself is a good example -- it is called via `ActiveRecord::Migration.check_pending!`, which is a class method.